x-pack/plugin/apm-integration/src/main/java/org/elasticsearch/xpack/apm/APMTracer.java

```diff
@@ -181,6 +231,22 @@ public void clear() {
             return List.copyOf(capturedSpans);
         }
 
+        public Stream<SpanData> findSpan(Predicate<SpanData> predicate) {
+            return capturedSpans.stream().filter(predicate);
```

Comment:
I believe it might be worth using `getCapturedSpans()` or `List.copyOf(capturedSpans)` to prevent concurrent modifications coming from stream and collector

Reply:
Sure, I'll make the change.